File: lib/ts/src/internal/types/VAny.ts
import { VTypeDeclaration } from './VTypeDeclaration';
import { ValidationFailure } from '../validation/ValidationFailure';
import { VType } from './VType';
import { generateRandomAny } from '../generation/GenerateRandomAny';
import { GenerateContext } from '../generation/GenerateContext';
import { ValidateContext } from '../validation/ValidateContext';

const anyName = 'Any';

export class VAny extends VType {
    getTypeParameterCount(): number {
        return 0;
    }

    validate(value: any, typeParameters: VTypeDeclaration[], ctx: ValidateContext): ValidationFailure[] {
        return [];
    }

    generateRandomValue(
        blueprintValue: any,
        useBlueprintValue: boolean,
        typeParameters: VTypeDeclaration[],
        ctx: GenerateContext,
    ): any {
        return generateRandomAny(ctx);
    }

    getName(): string {
        return anyName;
    }
}

File: lib/java/src/main/java/io/github/telepact/internal/types/VAny.java
package io.github.telepact.internal.types;

import static io.github.telepact.internal.generation.GenerateRandomAny.generateRandomAny;

import java.util.List;

import io.github.telepact.internal.generation.GenerateContext;
import io.github.telepact.internal.validation.ValidateContext;
import io.github.telepact.internal.validation.ValidationFailure;

public class VAny implements VType {

    static final String _ANY_NAME = "Any";

    @Override
    public int getTypeParameterCount() {
        return 0;
    }

    @Override
    public List<ValidationFailure> validate(
            Object value, List<VTypeDeclaration> typeParameters, ValidateContext ctx) {
        return List.of();
    }

    @Override
    public Object generateRandomValue(Object blueprintValue, boolean useBlueprintValue,
            List<VTypeDeclaration> typeParameters, GenerateContext ctx) {
        return generateRandomAny(ctx);
    }

    @Override
    public String getName() {
        return _ANY_NAME;
    }
}
